Common/Core/PID/TPCPIDResponse.h

```diff
@@ -0,0 +1,173 @@
+
+  /// Gets the expected signal of the track
+  template <typename TrackType>
+  float GetExpectedSignal(const TrackType& track, const o2::track::PID::ID id) const;
+  /// Gets the expected resolution of the track
+  template <typename CollisionType, typename TrackType>
+  float GetExpectedSigma(const CollisionType& collision, const TrackType& trk, const o2::track::PID::ID id) const;
+  /// Gets the number of sigmas with respect the expected value
+  template <typename CollisionType, typename TrackType>
+  float GetNumberOfSigma(const CollisionType& collision, const TrackType& trk, const o2::track::PID::ID id) const;
+  /// Gets the deviation to the expected signal
+  template <typename TrackType>
+  float GetSignalDelta(const TrackType& trk, const o2::track::PID::ID id) const;
+  /// Gets relative dEdx resolution contribution due to relative pt resolution
+  float GetRelativeResolutiondEdx(const float p, const float mass, const float charge, const float resol) const;
+
+  void PrintAll() const;
+
+ private:
+  std::array<float, 5> mBetheBlochParams = {0.0320981, 19.9768, 2.52666e-16, 2.72123, 6.08092};
+  std::array<float, 2> mResolutionParamsDefault = {0.07, 0.0};
+  std::vector<double> mResolutionParams = {5.43799e-7, 0.053044, 0.667584, 0.0142667, 0.00235175, 1.22482, 2.3501e-7, 0.031585};
+  float mMIP = 50.f;
+  float mChargeFactor = 2.3f;
+  float mMultNormalization = 11000.;
+  bool mUseDefaultResolutionParam = true;
+  std::unique_ptr<TFormula> mSigmaParametrization{new TFormula("fSigmaParametrization", "sqrt(([0]**2)*x[0]+(([1]**2)*(x[2]*[5])*(x[0]/sqrt(1+x[1]**2))**[2])+x[2]*x[3]**2+([4]*x[4])**2 +((x[5]*[6])**2)+(x[5]*(x[0]/sqrt(1+x[1]**2))*[7])**2)")};
+
+  ClassDefNV(Response, 2);
+
+}; // class Response
+
+/// Get expected Signal of the measurement
+template <typename TrackType>
+inline float Response::GetExpectedSignal(const TrackType& track, const o2::track::PID::ID id) const
+{
+  const float bethe = mMIP * o2::tpc::BetheBlochAleph(track.tpcInnerParam() / o2::track::pid_constants::sMasses[id], mBetheBlochParams[0], mBetheBlochParams[1], mBetheBlochParams[2], mBetheBlochParams[3], mBetheBlochParams[4]) * std::pow((float)o2::track::pid_constants::sCharges[id], mChargeFactor);
+  return bethe >= 0.f ? bethe : 0.f;
+}
+
+/// Gets the expected resolution of the measurement
+template <typename CollisionType, typename TrackType>
+inline float Response::GetExpectedSigma(const CollisionType& collision, const TrackType& track, const o2::track::PID::ID id) const
+{
+  float resolution = 0.;
+  if (mUseDefaultResolutionParam) {
+    const float reso = track.tpcSignal() * mResolutionParamsDefault[0] * ((float)track.tpcNClsFound() > 0 ? std::sqrt(1. + mResolutionParamsDefault[1] / (float)track.tpcNClsFound()) : 1.f);
+    reso >= 0.f ? resolution = reso : resolution = 0.f;
+  } else {
+
+    const double ncl = 159. / track.tpcNClsFound(); //
+    const double p = track.tpcInnerParam();
```

Comment:
This should be a float to avoid conversion, this applies also to other variables.

Reply:
I wanted to use floats for these variables, but the TFormula takes only doubles and no floats. This is why I did not change it.

Reply:
Ah I understand, ok!